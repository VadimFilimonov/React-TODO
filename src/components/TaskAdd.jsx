import React from "react";

class TaskAdd extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      input: "",
    };
  }

  addTask = (event) => {
    event.preventDefault();
    const { input } = this.state;
    if (input) {
      this.props.addTask(input);
      this.setState({ input: "" });
    }
  };

  onChange = (event) => {
    this.setState({ input: event.target.value });
  };

  render() {
    return (
      <form onSubmit={this.addTask}>
        <div className="input-group">
          <input
            className="form-control"
            onChange={this.onChange}
            value={this.state.input}
          ></input>
          <div className="input-group-append">
            <button className="input-group-text">Add</button>
          </div>
        </div>
      </form>
    );
  }
}

export default TaskAdd;
