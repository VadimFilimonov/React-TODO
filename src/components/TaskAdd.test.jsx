import React from "react";
import { render } from "@testing-library/react";
import TaskAdd from "./TaskAdd.jsx";

test("TaskAdd 1", () => {
  const { asFragment } = render(
    <TaskAdd />
  );
  expect(asFragment()).toMatchSnapshot();
});
