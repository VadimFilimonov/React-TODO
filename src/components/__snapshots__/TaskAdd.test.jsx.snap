// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskAdd 1 1`] = `
<DocumentFragment>
  <form
    style="display: flex;"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
      style="flex-grow: 1;"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
      >
        What needs to be done?
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              What needs to be done?
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
            />
          </svg>
        </span>
        Add
      </span>
    </button>
  </form>
</DocumentFragment>
`;
